src/pages/RepoPage/FailedTestsTab/FailedTestsPage/SelectorSection/BranchSelector/BranchSelector.tsx

```diff
@@ -76,20 +76,30 @@ const BranchSelector = () => {
     )
   }
 
+  const sortedBranchList = overview?.defaultBranch
+    ? [
+        // Pins the default branch to the top of the list always, filters it from results otherwise
+        { name: overview.defaultBranch, head: null },
+        ...(branchList?.branches?.filter(
+          (branch) => branch.name !== overview.defaultBranch
+        ) ?? []),
+      ]
+    : (branchList?.branches ?? [])
```

Comment:
`m`: What are your thoughts on utilizing `Set` here? I think we'd be able to drop the `filter` which would help us skip iterating over the entire array on each render, and reduce the overall complexity.

Also, what do you think about memo'izing this `sortedBranchList`?

Docs for `Set` can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#description), if you're looking for more details.

Reply:
I think it's actually a similar amount of complexity with using Set vs. Array, both from a runtime front and the code readability front; wasn't too sure which you meant.

Runtime front both are still O(n) since you need to populate the set with N elements before doing a removal, sets also don't respect ordering out the both so we'd still need to do a conversion back to array after right? It may end up being slower actually.

Readability front, were you thinking something like this?

`
const branchSet = Set(...branchList.branches)
Set.remove({ name: overview.defaultBranch, head: null })

 { name: overview.defaultBranch, head: null },
       ...branchSet,
 `

Memoizing is a good idea though, I think it'd help with either the Set or the Array approach.

Reply:
I was more thinking along the lines of:

```
const branchSet = Set({ name: overview.defaultBranch, head: null }, ...branchList.branches)
```

items added later won't override former items

Reply:
> items added later won't override former items

ahh okay I didn't realize sets were orderedSets by default, didn't get your point originally but that makes sense. So we're looking at something like this then right? 

```
  const sortedBranchList = useMemo(() => {
    return branchList?.branches
      ? [
          // Pins the default branch to the top of the list always, filters it from results otherwise
          new Set([
            overview?.defaultBranch && {
              name: overview.defaultBranch,
              head: null,
            },
            ...branchList.branches,
          ]),
        ]
      : []
  }, [branchList?.branches, overview?.defaultBranch])
```

It's around the same complexity imo haha but since I already did that thought exercise I can make the change for ya later

Reply:
I lied it's actually this

```
  const sortedBranchList = useMemo(() => {
    if (!branchList?.branches) return []

    return [
      ...(overview?.defaultBranch
        ? [
            {
              name: overview.defaultBranch,
              head: null,
            },
          ]
        : []),
      ...branchList.branches,
    ]
  }, [branchList?.branches, overview?.defaultBranch])
```

&& operator isn't so kind

Reply:
Yeah actually this doesn't seem to work bc of the && operator, if you have an elegant solution happy to try it out otherwise I'll leave as is

Reply:
This is def not a blocker, we can go with this for the launch today and take a peak as a fast follow